frame/clearing-house/src/tests/instruments.rs

```diff
@@ -160,28 +153,55 @@ proptest! {
 			);
 		})
 	}
-}
 
-proptest! {
 	#[test]
-	fn unrealized_funding_is_nonzero_iff_cum_rates_not_equal(
-		market in any_market(),
-		market_id in any::<MarketId>(),
-		base_asset_amount in bounded_decimal(),
-		quote_asset_notional_amount in bounded_decimal(),
-		cum_funding_delta in bounded_decimal(),
+	fn unrealized_funding_sign_is_correct(
+		(base_amount, quote_amount) in prop_oneof![
+			Just((-10, -100)),
+			Just((0, 0)),
+			Just((10, 100))
+		],
+		cum_funding_delta in
+			prop_oneof![Just(-1), Just(0), Just(1)].prop_map(|n| FixedI128::from((n, 10))),
 	) {
-		ExtBuilder::default().build().execute_with(|| {
-			let position = Position {
-				market_id,
-				base_asset_amount,
-				quote_asset_notional_amount,
-				last_cum_funding: market.cum_funding_rate + cum_funding_delta
-			};
-
-			let result = <TestPallet as Instruments>::unrealized_funding(&market, &position).unwrap();
+		let position = Position {
+			market_id: 0,
+			base_asset_amount: base_amount.into(),
+			quote_asset_notional_amount: quote_amount.into(),
+			last_cum_funding: 0.into(),
+		};
+		let cum_funding_rate = position.last_cum_funding + cum_funding_delta;
+		let market = Market {
+			cum_funding_rate_long: cum_funding_rate,
+			cum_funding_rate_short: cum_funding_rate,
+			..Default::default()
+		};
 
-			assert_eq!(cum_funding_delta.is_zero(), result.is_zero());
+		ExtBuilder::default().build().execute_with(|| {
+			let unrealized_funding = <TestPallet as Instruments>::unrealized_funding(
+				&market, &position
+			).unwrap();
+
+			// Positive unrealized funding means the position receive funds
+			//
+			//     position sign | funding rate sign | unrealized funding sign
+			// ---------------------------------------------------------------
+			//                -1 |                -1 | -1
+			//                -1 |                 1 |  1
+			//                 1 |                -1 |  1
+			//                 1 |                 1 | -1
+			//                 - |                 0 |  0
+			//                 0 |                 - |  0
+			let direction = position.direction();
+			if cum_funding_delta.is_zero() || direction.is_none() {
+				assert!(unrealized_funding.is_zero());
+			} else  {
+				assert!(match direction.unwrap() {
+					Direction::Long =>
+						cum_funding_delta.is_positive() != unrealized_funding.is_positive(),
+					_ => cum_funding_delta.is_positive() == unrealized_funding.is_positive(),
+				});
```

Comment:
What do you think about creating one test function for each case you are considering here?
The total number of random tests generated for this function would be split in 4 different non-related cases, but if you use one test function for each case you would be assessing more thoroughly each possibility.

Reply:
Yeah, that's true. It could be the case that the randomly generated inputs don't cover the edge case of `0` too. I'll split it

Reply:
Actually, I manually set the specific cases in the input strategies:
```rust
		(base_amount, quote_amount) in prop_oneof![
			Just((-10, -100)),
			Just((0, 0)),
			Just((10, 100))
		],
		cum_funding_delta in
			prop_oneof![Just(-1), Just(0), Just(1)].prop_map(|n| FixedI128::from((n, 10))),
```
So there's no real benefit of splitting it now